spec/controllers/vm_cloud_controller_spec.rb

```diff
@@ -179,10 +179,10 @@
       post :explorer
       expect(response.status).to eq(200)
       expect_any_instance_of(GtlHelper).to receive(:render_gtl).with match_gtl_options(
-        :model_name                     => 'ManageIQ::Providers::CloudManager::Vm',
+        :model_name                     => 'VmOrTemplate',
```

Comment:
This value is from: `@report_data_additional_options[:model]`
via -- (ApplicationHelper#model_to_report_data](https://github.com/ManageIQ/manageiq-ui-classic/blob/master/app/helpers/application_helper.rb#L171]

It is set in 

Locally, this passed in `x_active_tree = :instance_tree` and got the `model` back in [vm_common.rb](https://github.com/ManageIQ/manageiq-ui-classic/blob/master/app/controllers/vm_common.rb#L870)

Reply:
> So this was passing for me.

With rails 7 or 6.1?

FYI, ui-classic was failing with cross repo + rails 7 without this change.

Reply:
ok, so this calls `post` 2 times: `explorer` and `x_button`.
In the `explorer` call, the `db`/`model` is `ManageIQ::Providers::CloudManager::Vm`
In the `x_button` the `db`/`model` is `VmOrTemplate`.

The `explorer` call sets the `@report_data_additional_options.model` to the current model, which is `ManageIQ::Providers::CloudManager::Vm`.
The `x_button` call will reuse the `@report_data_additional_options.model` value if it is present.

In 6.1, the `x_button` has the instance variable set, so the `model` is `ManageIQ::Providers::CloudManager::Vm`
In 7.0, `actionpack-7.0.8.4/lib/action_controller/metal/testing.rb`
has a method `clear_instance_variables_between_requests`. So `@report_data_additional_options` is not set and it sets the `model` to the current model, which is `VmOrTemplate`.

Conclusions
1. The test may be setup incorrectly and the `model` should be the same for both calls. If this is a bug, it was probably always present in the test but did not rear it's head
2. The test framework changed, this may be just a testing change.
3. The expectation for instance variables in controllers may have changed and this may pose a problem in production.

Reply:
I agree, this "could" have some impact on our existing UI code.  I see get_view and get_db_view processing the show list options if this is not set.  If we relied on subsequent requests to have these values persisted over requests, we will likely set rebuilding the options with possibly new/different values on subsequent requests.  cc @GilbertCherrie @Fryguy.  Great find @kbrock.  I've been stumped looking at this one the past few several days off and on.

https://github.com/ManageIQ/manageiq-ui-classic/blob/43566cf0fdf4259db9384c959dcf72d99d2ac0c5/app/controllers/application_controller.rb#L1296

https://github.com/ManageIQ/manageiq-ui-classic/blob/43566cf0fdf4259db9384c959dcf72d99d2ac0c5/app/controllers/application_controller.rb#L1099

Reply:
Going forward, I **think**, this code is the right code but we'll have to merge this with the rails 7 branch as this will not pass on 6.1

Reply:
Note, I'll summarize Keenan's findings and put it in the commit message as I will forget about the details in a few days or even hours.

Reply:
@jrafanie Did you update the commit message?  Just waiting for this before merge.